src/app/components/containers/editor.rs

```diff
@@ -70,11 +76,24 @@ impl Editor {
 }
 
 pub struct EditorProps<'a> {
-    task: Option<&'a Task>,
+    stage: EditorStage,
+    id: String,
+    content: &'a String,
+    description: &'a String,
```

Comment:
These fields duplicate a lot of code and more fields will be added. Let's try to make an array of fields and then use array maps to render and modify the fields.